### PR TITLE
add in-memory database server to make testing easier

### PR DESCRIPTION
**Pull request by sd1li**

### CODE REVIEW

**Review by poundifdef — APPROVED** [review]

File: pkg/destinations/memory/memory.go [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+	table := ""
+	selectFrom := selectStarFromPat.FindStringSubmatch(query)
+	switch {
+	case len(selectFrom) == 2:
+		table = selectFrom[1]
+		rows = append(rows, tables.m[table]...)
+	case m.UseAllCaps:
+		rows = append(rows, json.RawMessage(`{"HELLO":"WORLD"}`))
+	default:
+		rows = append(rows, json.RawMessage(`{"hello":"world"}`))
```
[the remark this answers is not in the document]
Comment (sd1li) [reply pkg/destinations/memory/memory.go:85]: Behavior copied from `DummyDBServer`

[the remark this answers is not in the document]
Comment (chumaumenze) [reply pkg/destinations/memory/memory.go:85]: @poundifdef What is the usecase for the Dummy* structs?

[the remark this answers is not in the document]
Comment (poundifdef) [reply pkg/destinations/memory/memory.go:85]: They exist to let the code compile and run while we build the actual implementations of these components. They are meant to be used for debugging.

[the remark this answers is not in the document]
Comment (sd1li) [reply pkg/destinations/memory/memory.go:85]: @chumaumenze You can achieve that by calling `InsertBatchFromNDJson(tableName, r)`, then get the data back by calling `QueryJSON(`select * from tableName`, w)`. The only caveat is that you must use that specific query format and prolly a unique tableName.

**Pull request merged**